### PR TITLE
CSPL-1419: use different App Source for every CR for M4 test cases

### PR DESCRIPTION
**Pull request by jambrosiano**
Added different AppFramework sources for each CR(IDXC and SHC)

Passed tests:
[1] • [SLOW TEST:1660.371 seconds]
[1] m4appfw test
[1] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:30
[1]   Multisite Indexer Cluster with Search Head Cluster (m4) with App Framework
[1]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:298
[1]     integration, m4, appframework: can deploy a M4 SVA with App Framework enabled, install apps and downgrade them
[1]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:299
[1] ------------------------------
[1] • [SLOW TEST:980.434 seconds]
[1] m4appfw test
[1] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:30
[1]   Multi Site Indexer Cluster with Search Head Cluster (m4) with App Framework)
[1]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:764
[1]     integration, m4, appframework: can deploy a M4 SVA and have apps installed locally on Cluster Manager and Deployer
[1]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:765
[1] ------------------------------
[1] {"level":"info","ts":1637897575.467571,"msg":"testenv deleted.\n","testenv":"m4appfw-un4"}
[1] {"level":"info","ts":1637897575.467602,"msg":"testenv deleted.\n","testenv":"m4appfw-un4"}
[1]
[1] JUnit report was created: /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/m4appfw-un4_junit.xml
[1]
[1] Ran 2 of 2 Specs in 2657.405 seconds
[1] SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
[1] PASS
[2] • [SLOW TEST:1476.361 seconds]
[2] m4appfw test
[2] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:30
[2]   Multisite Indexer Cluster with Search Head Cluster (m4) with App Framework
[2]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:61
[2]     smoke, m4, appframework: can deploy a M4 SVA with App Framework enabled, install apps and upgrade them
[2]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:62
[2] ------------------------------
[2] • [SLOW TEST:1889.689 seconds]
[2] m4appfw test
[2] /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:30
[2]   Multisite Indexer Cluster with Search Head Cluster (m4) with App Framework
[2]   /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:536
[2]     integration, m4, appframework: can deploy a M4 SVA with App Framework enabled, install apps, scale up clusters, install apps on new pods, scale down
[2]     /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/appframework_test.go:537
[2] ------------------------------
[2] {"level":"info","ts":1637898299.7345579,"msg":"testenv deleted.\n","testenv":"m4appfw-b98"}
[2] {"level":"info","ts":1637898299.7345998,"msg":"testenv deleted.\n","testenv":"m4appfw-b98"}
[2]
[2] JUnit report was created: /Users/jambrosiano/splunk-operator/develop/splunk-operator/test/m4/appframework/m4appfw-b98_junit.xml
[2]
[2] Ran 2 of 2 Specs in 3381.666 seconds
[2] SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
[2] PASS

### CODE REVIEW

File: test/testenv/deployment.go [hunk 1]
```diff
@@ -683,7 +683,7 @@ func (d *Deployment) DeploySingleSiteClusterWithGivenAppFrameworkSpec(name strin
 }
 
 // DeployMultisiteClusterWithSearchHeadAndAppFramework deploys cluster-manager, indexers in multiple sites (SHC LM Optional) with app framework spec
-func (d *Deployment) DeployMultisiteClusterWithSearchHeadAndAppFramework(name string, indexerReplicas int, siteCount int, appFrameworkSpec enterpriseApi.AppFrameworkSpec, shc bool, mcName string, licenseMaster string) error {
+func (d *Deployment) DeployMultisiteClusterWithSearchHeadAndAppFramework(name string, indexerReplicas int, siteCount int, appFrameworkSpecIdxc enterpriseApi.AppFrameworkSpec, appFrameworkSpecShc enterpriseApi.AppFrameworkSpec, shc bool, mcName string, licenseMaster string) error {
 
 	// If license file specified, deploy License Manager
 	if d.testenv.licenseFilePath != "" {
```
Comment (pdhanoyasplunk) [inline test/testenv/deployment.go:689]: Unrelated change. Need to edit like 691 to `_, err := d.DeployLicenseManager(licenseMaster)`

File: test/m4/appframework/appframework_test.go [hunk 2]
```diff
@@ -85,20 +90,9 @@ var _ = Describe("m4appfw test", func() {
...
-			}
-			appSourceNameMC := "appframework-mc-" + testenv.RandomDNSName(3)
-			appSourceSpecMC := []enterpriseApi.AppSourceSpec{testenv.GenerateAppSourceSpec(appSourceNameMC, s3TestDirMC, appSourceDefaultSpecMC)}
-			appFrameworkSpecMC := enterpriseApi.AppFrameworkSpec{
-				Defaults:             appSourceDefaultSpecMC,
-				AppsRepoPollInterval: 60,
-				VolList:              volumeSpecMC,
-				AppSources:           appSourceSpecMC,
-			}
+			appSourceNameMC := "appframework-" + enterpriseApi.ScopeLocal + "mc-" + testenv.RandomDNSName(3)
```
Comment (pdhanoyasplunk) [inline test/m4/appframework/appframework_test.go:93]: This can be moved to BeforeEach.

File: test/m4/appframework/appframework_test.go [hunk 3]
```diff
@@ -85,20 +90,9 @@ var _ = Describe("m4appfw test", func() {
...
+			appSourceVolumeNameMC := "appframework-test-volume-mc-" + testenv.RandomDNSName(3)
```
Comment (pdhanoyasplunk) [inline test/m4/appframework/appframework_test.go:94]: Same.This can be moved to BeforeEach.

**Review by pdhanoyasplunk — CHANGES_REQUESTED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Review by tpereirasplunk — APPROVED** [review]

File: test/m4/appframework/appframework_test.go [hunk 4]
```diff
@@ -85,20 +94,9 @@ var _ = Describe("m4appfw test", func() {
...
```
Comment (pdhanoyasplunk) [inline test/m4/appframework/appframework_test.go:98]: Rename `appSourceVolumeNameMC` to `volumeNameMC`

File: test/m4/appframework/appframework_test.go [hunk 5]
```diff
@@ -317,24 +325,13 @@ var _ = Describe("m4appfw test", func() {
...
```
Comment (pdhanoyasplunk) [inline test/m4/appframework/appframework_test.go:333]: Rename `appSourceVolumeNameMC` to `volumeNameMC`

**Review by pdhanoyasplunk — APPROVED** [review]

**Pull request merged**